src/mito2/src/cache/index.rs

```diff
@@ -59,25 +60,47 @@ where
         offset: u64,
         size: u32,
     ) -> index::inverted_index::error::Result<Vec<u8>> {
-        let range = offset as usize..(offset + size as u64) as usize;
-        if let Some(cached) = self.cache.get_index(IndexKey {
-            file_id: self.file_id,
-        }) {
-            CACHE_HIT.with_label_values(&[INDEX_CONTENT_TYPE]).inc();
-            Ok(cached[range].to_vec())
-        } else {
-            let mut all_data = Vec::with_capacity(1024 * 1024);
-            self.inner.read_all(&mut all_data).await?;
```

Comment:
`read_all` can be removed.

Reply:
these functions are trait member of `InvertedIndexReader` and used for `InvertedIndexBlobReader` as well, do we need to change them all?